pkg/frontend/querymiddleware/limits_test.go

```diff
@@ -186,6 +185,103 @@ func TestLimitsMiddleware_MaxQueryLookback(t *testing.T) {
 	}
 }
 
+func TestLimitsMiddleware_MaxQueryLookback_InstantQuery(t *testing.T) {
+	const (
+		thirtyDays = 30 * 24 * time.Hour
+		sixtyDays  = 60 * 24 * time.Hour
+	)
+
+	now := time.Now()
+
+	tests := map[string]struct {
+		maxQueryLookback      time.Duration
+		blocksRetentionPeriod time.Duration
+		reqTime               time.Time
+		expectedSkipped       bool
+		expectedTime          time.Time
+	}{
+		"should not manipulate request time if maxQueryLookback and blocksRetentionPeriod are both disabled": {
```

Comment:
NitX3: If I understand correctly, start==end for instant query so either we allow it or not, never manipulate it so this is kind of weird name for the testcase. 

```suggestion
		"should allow executing a request if maxQueryLookback and blocksRetentionPeriod are both disabled": {
```

Reply:
Yes, absolutely. Going to fix the descriptions in this test.

Reply:
Done in https://github.com/grafana/mimir/pull/8410/commits/7fac98107fcc2d8abfbe42729ed0c3d45312059d